### ISSUE TITLE
ChartJS: Implement linear, logarithmic and time series charts

### ISSUE THREAD

**Issue by sandsm** [primefaces/primefaces#5131]
## 1) Environment
- PrimeFaces version: 7.0
- Does it work on the newest released PrimeFaces version? No
- Does it work on the newest sources in GitHub? Unsure
- Application server + version: Wildfly 17
- Affected browsers: All

## 2) Expected behavior
It would be nice to have access to linear logarithmic and time series charts in chartjs from PF.

## 3) Actual behavior
It seems category charts are the only charts implemented in the PF models.

## 4) Steps to reproduce
Search web and code base for evidence of x,y data set to no avail.

## 5) Sample XHTML
None

## 6) Sample bean
None

**Comment by melloware** [comment]
Do you mind pointing out some examples from the charts js samples that currently can be done so I can investigate?

https://www.chartjs.org/samples/latest/

**Comment by sandsm** [comment]
Thanks for the quick reply @melloware.

I included a linear scale one I found on codepen here as the samples don't have an example of linear scaling.
https://codepen.io/thohoh/pen/aVoYLq

These time ones below are probably useful.
https://www.chartjs.org/samples/latest/scales/time/line.html
https://www.chartjs.org/samples/latest/scales/time/line-point-data.html
https://www.chartjs.org/samples/latest/scales/logarithmic/scatter.html

Cheers,
Mike.

**Comment by melloware** [comment]
So for your first one (linear scale) this is already fixed in 7.1 with this ticket: https://github.com/primefaces/primefaces/issues/4564

You will be able to do:
```java
List<Object> data = new ArrayList<>();
data.add(new NumericPoint(1,3));
data.add(new NumericPoint(2,4));
data.add(new NumericPoint(8,12));
```

**Comment by melloware** [comment]
As for the logarithmic example (https://www.chartjs.org/samples/latest/scales/logarithmic/scatter.html) I believe that is also possible.

See: https://github.com/primefaces/primefaces/blob/master/src/main/java/org/primefaces/model/charts/axes/cartesian/linear/CartesianLinearAxes.java

Just set the type `axes.setType("logarithmic");`

**Comment by sandsm** [comment]
Thanks @melloware for the great service.

**Comment by melloware** [comment]
@mores that will be fixed in 14.0.0 you can see this discussion here: https://github.com/primefaces/primefaces/pull/11291

we are creating a new more flexible "p:chart" for 14.0.0 that gives you complete control over ChartJS rendering and allows for DateTime values and we even include it in one of our examples in the 14.0.0 showcase.
